frontend/__tests__/components/language-switcher.test.tsx

```diff
@@ -0,0 +1,46 @@
+import { render, screen, waitFor } from '@testing-library/react';
+
+import { createRemixStub } from '@remix-run/testing';
+
+import { axe, toHaveNoViolations } from 'jest-axe';
+
+import { LanguageSwitcher } from '~/components/language-switcher';
+
+expect.extend(toHaveNoViolations);
+
+vi.mock('~/utils/env.ts', () => {
+  const getClientEnv = vi.fn();
+  getClientEnv.mockReturnValue({
+    LANG_QUERY_PARAM: 'lang',
+  });
+  return {
+    getClientEnv,
+  };
+});
+
+vi.mock('~/utils/locale-utils.ts', () => {
+  const getAltLanguage = vi.fn();
+  getAltLanguage.mockReturnValue('en');
+  return {
+    getAltLanguage,
+  };
+});
+
+describe('Language Switcher', () => {
+  it('Should render and not have a11y issues', async () => {
+    const RemixStub = createRemixStub([
+      {
+        Component: () => <LanguageSwitcher>English</LanguageSwitcher>,
```

Comment:
**Nitpick:**

```suggestion
        Component: () => <LanguageSwitcher>Français</LanguageSwitcher>,
```
<br/>

**Reasoning:** it could be confusing to future devs to see `English` with the toggle href being `?lang=en`. It confused me for a minute.